### PR TITLE
Zmq gst camera

### PR DESCRIPTION
**Pull request by jaybdub**
- zmq camera_publisher  ``scripts/camera_publisher.py``
- hardware accelerated gst jpeg encoding ``bgr8_to_jpeg_gst``
- zmq camera subscriber

### CODE REVIEW

File: scripts/camera_publisher.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        sample = appsink.emit('pull-sample')
+        buf = sample.get_buffer()
+        caps = sample.get_caps()
+        height = caps.get_structure(0).get_value('height')
+        width = caps.get_structure(0).get_value('width')
+        (result, mapinfo) = buf.map(Gst.MapFlags.READ)
+
+        image = np.ndarray(
+            shape=(height, width, 3),
+            buffer=buf.extract_dup(0, buf.get_size()),  # extract_dup to copy
```
Comment (jaybdub) [inline scripts/camera_publisher.py:71]: We could probably just publish here, and avoid an extra copy :|

**Comment by tokk-nv** [comment]
Can you edit the create-sdcard-image-from-scratch.sh to indicate what needs to be installed for this?

**Comment by jaybdub** [comment]
> Can you edit the create-sdcard-image-from-scratch.sh to indicate what needs to be installed for this?

Sure i'll check for the JpegEncoder.

Scripts/camera_publisher.py / new Camera class should work with existing setup.

scripts/camera_publisher.py should actually work with base L4T I think.  (maybe some small python requirements) This way it could be run outside container without issue.

**Comment by jaybdub** [comment]
> Can you edit the create-sdcard-image-from-scratch.sh to indicate what needs to be installed for this?

Done

**Pull request merged**